File: ve/src/lib/Memory.ts
//ローカルストレージの読み書き(Jsonとの相互変換対応)
export default class Memory{
	//読込
	public static load<T>(key:string):T{
		return JSON.parse(localStorage[key]);
	}

	//保存
	public static save<T>(key:string,memory:T):void{
		localStorage.setItem(key,JSON.stringify(memory));
	}

	//削除
	public static remove<T>(key:string):void{
		localStorage.removeItem(key);
	}
}

File: br/Lib/Memory.cs
//ローカルストレージの読み書き(Jsonとの相互変換対応)
using System.Threading.Tasks;
using System.Text.Json;
using Microsoft.JSInterop;

class Memory{
	//読込
	public static async Task<T> load<T>(IJSRuntime js,string key){
		return JsonSerializer.Deserialize<T>(await js.InvokeAsync<string>(
			"localStorage.getItem",new[]{key}));
	}

	//保存
	public static async Task save<T>(IJSRuntime js,string key,T memory){
		await js.InvokeVoidAsync(
			"localStorage.setItem",new[]{key,JsonSerializer.Serialize<T>(memory)});
	}

	//削除
	public static async Task remove(IJSRuntime js,string key){
		await js.InvokeVoidAsync(
			"localStorage.removeItem",new[]{key});
	}
}
